tests/utils/test_environment.py

```diff
@@ -50,3 +51,27 @@ def test_mlflow_conda_env_includes_pip_dependencies_and_pip_is_specified(pip_spe
         assert conda_dep in env["dependencies"]
     assert pip_specification in env["dependencies"]
     assert env["dependencies"].count("pip") == (2 if pip_specification == "pip" else 1)
+
+
+def _normalize_version(version):
+    """
+    Normalizes a package version (e.g. '1.5.0-dev0' -> '1.5.0.dev0')
+    """
+    from packaging.version import Version
+
+    return str(Version(version))
+
+
+def test_infer_pip_requirements():
+    import xgboost as xgb
+    import sklearn
+
+    model = xgb.XGBClassifier()
+
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(model, artifact_path="model")
+        model_uri = mlflow.get_artifact_uri("model")
+
+    pip_reqs = _infer_pip_requirements(model_uri)
+    assert f"xgboost=={_normalize_version(xgb.__version__)}" in pip_reqs
```

Comment:
The `METADATA` file of `xgboost 1.5.0-dev.0` (note `-` before `dev`) looks like this:


`importlib_metadata` reads a packge version from this file.

```python
>>> xgboost.__version__
1.5.0-dev.0
>>> importlib_metadata.version("xgboost")
1.5.0.dev0
```